streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -282,9 +282,8 @@ public void resume() {
                 break;
 
             case SUSPENDED:
-                // just re-initilaize the topology is sufficient
-                initializeTopology();
-
+                // just transit the state without any logical changes: suspended and restoring states
+                // are not actually any different for inner modules
```

Comment:
Seems like maybe we could just get rid of the `suspended` state entirely, but I guess we plan to do that anyway if/when we remove support for EAGER

Reply:
Yes -- from restoring to suspended it is only to update the local `checkpoint` file, from suspended to restoring there's actually nothing needed. In that sense restoring and suspended states are very similar already, BUT still not exactly the same.

I do hope we get rid of it some time after we remove the EAGER protocol.